### ISSUE TITLE
Persistent volumes in docker engine swarm mode across multiple nodes

### ISSUE THREAD

**Issue by viveky4d4v** [moby/moby#30912]
This isn't an issue but more of a direction required in order to share volumes across multiple nodes in docker engine swarm mode to achieve data persistence .
There are very few docs available to achieve this in swarm mode .

**Comment by pascalandy** [comment]
Docker is not support this natively. So at the moment I use Resilio Sync. GlusterFS and Infinit were not stable for me. Rex-Ray seems promising but they are clear that this is not prod ready. 

I'm still looking to use Minio in the mix https://twitter.com/_pascalandy/status/830132966470873089

Good luck!

**Comment by alexellis** [comment]
Any official advise on this topic @justincormack ?

**Comment by RotemEmergi** [comment]
+1

**Comment by justincormack** [comment]
I don't think we have official advice. If you have stability issues with Infinit I am sure they would like to know - I have not when I have used it.

**Comment by sitamet** [comment]
I'm currently developing a[ Ceph RBD plugin managed using Docker Engine](https://github.com/wetopi/docker-volume-rbd) plugin system.
Works with docker swarm mode.

**Comment by pascalandy** [comment]
@justincormack I was in touch with the Infinit team for 5 straight days as I was enable to follow the doc. I wrote my conclusion here: https://github.com/pascalandy/infinit_sh-storage

It worked for few days but the system crashed and I let go. 

Then I extensively use Minio and it was stable for few days :) 
I wrote my conclusion here: http://blog.pascalandy.com/the-complete-guide-to-attach-a-docker-volume-with-minio-on-your-docker-swarm-cluster/ 

--- 

My **challenge** at this point: 
Let say I create a docker volume:

```
docker volume create -d minio/minfs \  
--name bucket-dev-e \
-o endpoint=http://ip10_0_23_3-9001.play-with-docker.com/ \
-o access-key=A5a0a87b725552daXd \
-o secret-key=369f5e7b4a41e25452c353D629a24c372b62c90 \
-o bucket=bucket-dev-e
```

By default we cannot specify subpaths (i.e.e bucket-dev-e/ghost/site2/images) within a Docker Volume. I don't want to use one Docker Volume for each of the 100x (potentially 1000x) sites I’m hosting.

Cheers!
